sklearn/metrics/_scorer.py

```diff
@@ -102,11 +102,15 @@ def __call__(self, estimator, *args, **kwargs):
         cached_call = partial(_cached_call, cache)
 
         for name, scorer in self._scorers.items():
-            if isinstance(scorer, _BaseScorer):
-                score = scorer._score(cached_call, estimator, *args, **kwargs)
-            else:
-                score = scorer(estimator, *args, **kwargs)
-            scores[name] = score
+            try:
+                if isinstance(scorer, _BaseScorer):
+                    score = scorer._score(cached_call, estimator, *args, **kwargs)
+                else:
+                    score = scorer(estimator, *args, **kwargs)
+                scores[name] = score
+            except Exception as e:
+                scores[name] = e
```

Comment:
I know I recommended this approach, but is there any Python specific issues with saving the Exception like this?

REF: https://stackoverflow.com/questions/8238360/how-to-save-traceback-sys-exc-info-values-in-a-variable

If there is an issue, we would need to try another approach.

Reply:
Based on the comments to the Stackoverflow question ("Be careful when you take the exception object or the traceback object out of the exception handler, since this causes circular references and gc.collect() will fail to collect. "), I agree, we should avoid it. 

If we choose to go that way, we have to store the type of the exception, the text and the `__traceback__`. Im not sure however how can we raise a new exception using data from a previous one.

Reply:
I'm okay with raising an exception in `_MultimetricScorer` and a new bool parameter in `__init__`.

> In that case we should only store the formatted exception text to use it in the warning.

In the warning situation, how do you want to pass the formatted exception text back up to `_score`?

Reply:
Passing it as a string. And then in the `_score` function in the `scores.items()` we could look for strings, which would indicate the exception.

```
except Exception as e:
    if raise_exp:
        raise e
    else:
        scores[name] = format_exc()
```

Reply:
Let's go with a string for now. I do not see a great alternative that is less complicated.